### DISCUSSION TITLE
Access to OTGW via SSH?

### DISCUSSION THREAD

**Discussion by tonga1968** [rvdbreemen/OTGW-firmware#76] [Q&A] [answered]
Hello,

I  have a Raspberry Pi, which is only accessible with a bash terminal over a SSH tunnel. The house, that I need to control, is about 120km away from me.
Is there any way to reach the OTGW on this way (and communicate with it)?
If so, how can I achieve this?
Otherwise, I would like to suggest this for future versions – perhaps with ncurses or something in this manner. :-)

Thank you
Christoph

**Comment by DaveDavenport** [comment, marked as answer]
I doubt that adding a ssh server is  feasible on the esp8266.  we need the encryption, user accounting and a shell.
I also do not think it is wise to have OTGW connected up to the internet.
Its better to connect to f.e. the web interface via the pi.

**Reply by tonga1968** [reply]
Oh, that seems to be a misunderstanding: My question is not how to connect to the OTGW via SSH directly, but how to talk with the otmonitor, which shall be installed on the pi. The SSH server is already there!
I can log in to the pi via ssh, but I cannot use a web interface, because I only have access to the pi over a text terminal (bash).
BTW: I have seen, that my OTGW device (which I bought on nodo-shop) has a I²C bus. Can I use this to connect to the pi?

**Reply by DaveDavenport** [reply]
with ssh you can use f.e. localforward (https://www.ssh.com/academy/ssh/tunneling/example) to connect to the web interface.

**Reply by emkookmer** [reply]
Another alternative to SSH tunnel would be a VPN tunnel.
You can run Docker on your PI and with it create a VPN server container.
https://hub.docker.com/search?q=Openvpn&type=image

**Reply by rvdbreemen** [reply]
@tonga1968 now why would you wanna use otmonitor on a pi? As my firmware does all the decoding otmonitor does. So you no longer need a separate otmonitor on a pi. That's what got my started on the first place. The esp8266 is powerfull enough to do otmonitor the translation of all OpenTherm messages to MQTT. That way you can integrate it with Domoticz or Home Assistant (my favorite btw).

I have setup my edge router to accept a secure VPN (based on openvpn) and thru that I connect directly to my firmware based on either the WebUI or MQTT or telnet. No SSH needed my setup.
